bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyScriptEngineConfiguration.java

```diff
@@ -0,0 +1,296 @@
+            "gem_home");
+
+    private final static Path DEFAULT_RUBYLIB = Paths.get(OpenHAB.getConfigFolder(), "automation", "lib", "ruby");
+
+    private final static String GEM_HOME = "gem_home";
+
+    // Map of configuration parameters
+    private final static Map<String, OptionalConfigurationElement> CONFIGURATION_PARAMETERS = Map.ofEntries(
+            Map.entry("local_context",
+                    new OptionalConfigurationElement.Builder(OptionalConfigurationElement.Type.SYSTEM_PROPERTY)
+                            .mappedTo("org.jruby.embed.localcontext.scope").defaultValue("singlethread").build()),
+
+            Map.entry("local_variable",
+                    new OptionalConfigurationElement.Builder(OptionalConfigurationElement.Type.SYSTEM_PROPERTY)
+                            .mappedTo("org.jruby.embed.localvariable.behavior").defaultValue("transient").build()),
+
+            Map.entry(GEM_HOME,
+                    new OptionalConfigurationElement.Builder(OptionalConfigurationElement.Type.RUBY_ENVIRONMENT)
+                            .mappedTo("GEM_HOME").defaultValue(DEFAULT_GEM_HOME.toString()).build()),
+
+            Map.entry("rubylib",
+                    new OptionalConfigurationElement.Builder(OptionalConfigurationElement.Type.RUBY_ENVIRONMENT)
+                            .mappedTo("RUBYLIB").defaultValue(DEFAULT_RUBYLIB.toString()).build()),
+
+            Map.entry("gems", new OptionalConfigurationElement.Builder(OptionalConfigurationElement.Type.GEM).build()));
+
+    private final static Map<OptionalConfigurationElement.Type, List<OptionalConfigurationElement>> CONFIGURATION_TYPE_MAP = CONFIGURATION_PARAMETERS
+            .values().stream().collect(Collectors.groupingBy(v -> v.type));
+
+    /**
+     * Update configuration
+     * 
+     * @param config Configuration parameters to apply to ScriptEngine
+     * @param factory ScriptEngineFactory to configure
+     */
+    void update(Map<String, Object> config, ScriptEngineFactory factory) {
+        logger.trace("JRuby Script Engine Configuration: {}", config);
+        config.forEach(this::processConfigValue);
+        configureScriptEngine(factory);
+    }
+
+    /**
+     * Apply configuration key/value to known configuration parameters
+     * 
+     * @param key Configuration key
+     * @param value Configuration value
+     */
+    private void processConfigValue(String key, Object value) {
+        OptionalConfigurationElement configurationElement = CONFIGURATION_PARAMETERS.get(key);
+        if (configurationElement != null) {
+            configurationElement.setValue(value.toString());
+        } else {
+            logger.debug("Ignoring unexpected configuration key: {}", key);
+        }
+    }
+
+    /**
+     * Configure the ScriptEngine
+     * 
+     * @param factory Script Engine to configure
+     */
+    void configureScriptEngine(ScriptEngineFactory factory) {
+
```

Comment:
```suggestion
```

Please also spend a couple of minutes to fix the other trivial static analysis findings so the code is nice and clean.

See: https://ci.openhab.org/job/PR-openHAB-Addons/6411/Static_20Code_20Analysis_20Report/